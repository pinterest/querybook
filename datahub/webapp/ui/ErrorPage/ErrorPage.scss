@import './../../scss_variables.scss';

.ErrorPage {
    position: relative;
    .ErrorPage-message {
        @include ellipsis(30);
        white-space: pre-wrap;
    }
}
