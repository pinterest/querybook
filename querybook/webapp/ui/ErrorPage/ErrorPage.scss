@import './../../scss_variables.scss';

.ErrorPage {
    position: relative;
    min-height: 200px;
    .ErrorPage-message {
        @include ellipsis(30);
        white-space: pre-wrap;
    }
}
