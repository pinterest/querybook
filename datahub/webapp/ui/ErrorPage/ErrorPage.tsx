import React from 'react';
import classNames from 'classnames';

import history from 'lib/router-history';
import { Center } from 'ui/Center/Center';
import { Title } from 'ui/Title/Title';
import { FullHeight } from 'ui/FullHeight/FullHeight';

import './ErrorPage.scss';

interface IErrorPageProps {
    className?: string;
    errorCode?: number;
    errorTitle?: string;
}

const httpCodeToString = {
    404: 'Not Found',
    403: 'Forbidden',
};

const httpCodeToDescription = {
    404: 'No match for',
    403: 'You are not allowed to access',
};

export const ErrorPage: React.FunctionComponent<IErrorPageProps> = ({
    className,
    children,
    errorCode,
    errorTitle,
}) => {
    if (!errorTitle) {
        errorTitle =
            errorCode in httpCodeToString
                ? `${errorCode}: ${httpCodeToString[errorCode]}`
                : 'Unknown Error';
    }

    const message = children ? (
        children
    ) : (
        <>
            {httpCodeToDescription[errorCode] || 'Unknown error occured at'}
            <code className="ErrorPage-path ml4">{location.pathname}</code>.
            <span className="ml4">
                Click <a onClick={history.goBack}>here</a> to go back.
            </span>
        </>
    );

    return (
        <FullHeight
            className={classNames({ ErrorPage: true, [className]: className })}
        >
            <Center>
                <Title size={2}>{errorTitle}</Title>
                <Title subtitle size={6} className="ErrorPage-message">
                    {message}
                </Title>
            </Center>
        </FullHeight>
    );
};
