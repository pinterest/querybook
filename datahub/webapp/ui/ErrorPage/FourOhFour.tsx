import React from 'react';

import { ErrorPage } from 'ui/ErrorPage/ErrorPage';

export const FourOhFour: React.FunctionComponent = ({ children }) => (
    <ErrorPage errorCode={404} errorMessage={children} />
);
