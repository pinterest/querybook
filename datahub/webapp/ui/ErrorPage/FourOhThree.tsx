import React from 'react';
import { ErrorPage } from './ErrorPage';

export const FourOhThree: React.FunctionComponent = ({ children }) => (
    <ErrorPage errorCode={403} errorMessage={children} />
);
