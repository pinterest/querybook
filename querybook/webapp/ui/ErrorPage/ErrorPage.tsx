import clsx from 'clsx';
import React from 'react';

import history from 'lib/router-history';
import { Center } from 'ui/Center/Center';
import { FullHeight } from 'ui/FullHeight/FullHeight';
import { Subtitle, Title } from 'ui/Title/Title';

import './ErrorPage.scss';

interface IErrorPageProps {
    className?: string;
    errorCode?: number;
    errorTitle?: string;
    errorMessage?: any;
}

const httpCodeToString = {
    404: 'Not Found',
    403: 'Forbidden',
};

const httpCodeToDescription = {
    404: 'No match for',
    403: 'You are not allowed to access',
};

export const ErrorPage: React.FunctionComponent<IErrorPageProps> = ({
    className,
    children,
    errorMessage,
    errorCode,
    errorTitle,
}) => {
    if (!errorTitle) {
        errorTitle =
            errorCode in httpCodeToString
                ? `${errorCode}: ${httpCodeToString[errorCode]}`
                : 'Unknown Error';
    }

    const message = errorMessage ? (
        errorMessage
    ) : (
        <>
            {httpCodeToDescription[errorCode] || 'Unknown error occured at'}
            <code className="ErrorPage-path ml4">{location.pathname}</code>.
            <span className="ml4">
                Click <a onClick={history.goBack}>here</a> to go back.
            </span>
        </>
    );

    const classNameProp = clsx({
        ErrorPage: true,
        [className]: className,
    });

    return (
        <FullHeight className={classNameProp}>
            <Center className="flex-column">
                <Title size="xxlarge" className="mb24">
                    {errorTitle}
                </Title>
                <Subtitle className="ErrorPage-message mb16">
                    {message}
                </Subtitle>
                {children}
            </Center>
        </FullHeight>
    );
};
