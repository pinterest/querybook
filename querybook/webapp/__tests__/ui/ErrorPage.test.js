import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { ErrorPage } from '../../ui/ErrorPage/ErrorPage';
import { FourOhFour } from '../../ui/ErrorPage/FourOhFour';
import { FourOhThree } from '../../ui/ErrorPage/FourOhThree';

it('renders without crashing', () => {
    shallow(<ErrorPage />);
});
it('renders without crashing', () => {
    shallow(<FourOhFour />);
});
it('renders without crashing', () => {
    shallow(<FourOhThree />);
});

describe('matches enzyme snapshots', () => {
    it('matches snapshot', () => {
        const wrapper = shallow(<ErrorPage />);
        const serialized = toJson(wrapper);
        expect(serialized).toMatchSnapshot();
    });
    it('matches snapshot', () => {
        const wrapper = shallow(<FourOhFour />);
        const serialized = toJson(wrapper);
        expect(serialized).toMatchSnapshot();
    });
    it('matches snapshot', () => {
        const wrapper = shallow(<FourOhThree />);
        const serialized = toJson(wrapper);
        expect(serialized).toMatchSnapshot();
    });
});
