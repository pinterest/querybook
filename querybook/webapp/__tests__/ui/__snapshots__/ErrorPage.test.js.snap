// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches enzyme snapshots matches snapshot 1`] = `
<FullHeight
  className="ErrorPage"
>
  <Center
    className="flex-column"
  >
    <Title
      className="mb24"
      gap={false}
      size={2}
      subtitle={false}
    >
      Unknown Error
    </Title>
    <Title
      className="ErrorPage-message mb16"
      gap={false}
      size={6}
      subtitle={true}
    >
      Unknown error occured at
      <code
        className="ErrorPage-path ml4"
      >
        /
      </code>
      .
      <span
        className="ml4"
      >
        Click 
        <a
          onClick={[Function]}
        >
          here
        </a>
         to go back.
      </span>
    </Title>
  </Center>
</FullHeight>
`;

exports[`matches enzyme snapshots matches snapshot 2`] = `
<ErrorPage
  errorCode={404}
/>
`;

exports[`matches enzyme snapshots matches snapshot 3`] = `
<ErrorPage
  errorCode={403}
/>
`;
